router/index.js

```diff
@@ -24,20 +31,102 @@ router.post('/login', (req, res, next) => {
 
 // get list of cameras
 
-router.get('/camera', (req, res, next) => {
-  res.sendStatus(200);
+router.get('/camera', async (req, res, next) => {
+
+  try {
+    const allcameras = await db.getAllCameras();
+
+    if (allcameras.length === 0) {
+      res.json({message: 'no cameras available'});
+    } else {
+      res.send(allcameras);
+    }
+
+
+  } catch (error) {
+    next(error);
+    return; // Ensures rest of function doesn't run
+  }
 });
 
 // get camera by ID
 
-router.get('/camera/:id', (req, res, next) => {
-  res.sendStatus(200);
+router.get('/camera/:id', async (req, res, next) => {
+  // check if id is an integer
+  if (!Number.isInteger(parseInt(req.params.id))) {
+    res
+      .status(400)
+      .json({message:'ID needs to be an integer'})
+  } else {
+    db.getCameraById(req.params.id)
+      .then((camera) => {
+        if (camera == null) {
+          res
+            .status(400)
+            .json({message:'no camera found by ID specified'})
```

Comment:
Nice formatting/indenting here. This looks very clean.

Reply:
Thanks. using eslint in visual studio code really helps.